subsys/nrf_security/src/drivers/cracen/cracenpsa/src/sign.c

```diff
@@ -120,25 +117,8 @@ static int cracen_signature_prepare_ec_prvkey(struct si_sig_privkey *privkey, ch
 		return SX_ERR_INVALID_ARG;
 	}
 
-	if (IS_ENABLED(PSA_NEED_CRACEN_PURE_EDDSA_TWISTED_EDWARDS)) {
-		if (alg == PSA_ALG_PURE_EDDSA) {
-			privkey->def = si_sig_def_ed25519;
-			privkey->key.ed25519 = (struct sx_ed25519_v *)key_buffer;
-			return SX_OK;
-		}
-	}
-
-	if (IS_ENABLED(PSA_NEED_CRACEN_ED25519PH)) {
-		if (alg == PSA_ALG_ED25519PH) {
-			privkey->def = si_sig_def_ed25519ph;
-			privkey->key.ed25519 = (struct sx_ed25519_v *)key_buffer;
-			if (message) {
-				return cracen_signature_set_hashalgo(&privkey->hashalg, alg);
-			} else {
-				return cracen_signature_set_hashalgo_from_digestsz(
-					&privkey->hashalg, alg, digestsz);
-			}
-		}
+	if (alg == PSA_ALG_PURE_EDDSA || alg == PSA_ALG_ED25519PH) {
```

Comment:
Was that intentional to remove the IS_ENABLED statement which requires the relevant PSA_NEED to be enabled here?

Reply:
Yes, but actually this statement is not needed either anymore. `cracen_signature_prepare_ec_prvkey` is never called while using eddsa, as `cracen_signature_prepare_ec_prvkey` is a support function for sicrypto